packages/studiocms/src/routes/api/dashboard/content/page.ts

```diff
@@ -66,6 +66,82 @@ function getParentFolderValue(value?: string) {
 	return value;
 }
 
+function isValidString(value: unknown): value is string {
+	return typeof value === 'string' && value.trim() !== '';
+}
+
+function validString(field: FormDataEntryValue | null): string | undefined {
+	if (!isValidString(field)) {
+		return undefined;
+	}
+	return field.toString();
+}
+
+function validateStringField(field: FormDataEntryValue | null, fieldName: string): string {
+	if (!isValidString(field)) {
+		throw new Error(`Invalid form data, ${fieldName} is required`);
+	}
+	return field.toString();
+}
+
+function validateSlugField(field: FormDataEntryValue | null, fieldName: string): string {
+	const slug = validateStringField(field, fieldName);
+
+	/**
+	 * Regex breakdown:
+	 *
+	 * ^[a-z0-9]+        : starts with one or more lowercase letters or numbers
+	 *
+	 * (?:-[a-z0-9]+)*  : followed by zero or more groups of a hyphen and one
+	 *                    or more lowercase letters or numbers
+	 *
+	 * $                : end of the string
+	 *
+	 * This ensures the slug is lowercase and only contains letters, numbers,
+	 * and hyphens, without consecutive hyphens or leading/trailing hyphens.
+	 */
+	const slugRegex = /^[a-z0-9]+(?:-[a-z0-9]+)*$/;
+
+	if (!slugRegex.test(slug)) {
+		throw new Error(
+			`Invalid form data, ${fieldName} must be lowercase and can only contain letters, numbers, and hyphens`
+		);
+	}
+
+	return slug;
+}
+
+function buildPageDataObject(
+	formData: FormData,
+	pageId?: string
+): UpdatePageData | { error: string } {
+	try {
+		const data: UpdatePageData = {
```

Comment:
```suggestion
        // TODO Fix validation logic. Move over to JSON. Use something like Zod here instead of throwing errors 🤮
		const data: UpdatePageData = {
```